tendrl/gluster_integration/sds_sync/__init__.py

```diff
@@ -66,6 +66,29 @@ def _run(self):
                             GlusterBrickDir()
         gluster_brick_dir.save()
 
+        try:
+            NS._int.client.read(
+                "clusters/%s/cluster_network" % NS.tendrl_context.integration_id
+            )
+        except etcd.EtcdKeyNotFound:
+            try:
+                node_networks = NS._int.client.read(
+                    "nodes/%s/Networks" % NS.node_context.node_id
+                )
+                # TODO: (team) this logic needs to change later
+                # multiple networks supported for gluster use case
+                node_network = NS.tendrl.objects.NodeNetwork(
+                    interface=node_networks.leaves.next().key.split('/')[-1]
+                ).load()
+                cluster = NS.tendrl.objects.Cluster(
+                    integration_id=NS.tendrl_context.integration_id
+                ).load()
+                cluster.cluster_network = node_network.subnet
```

Comment:
In case of gluster as recommendation single network would be available on nodes and yes in that case node_network.subnet should be equal to cluster.cluster_network.

UI uses the cluster.cluster_network value to filter the nodes which could be added to this cluster. If node's subnet matches the value then they can be added to the cluster.